download/download.go

```diff
@@ -97,30 +101,62 @@ func (d *Downloader) ClearCache() {
 
 // Start tells the Downloader to begin downloading bundles.
 func (d *Downloader) Start(ctx context.Context) {
-	go d.loop()
+	go d.doStart(ctx)
 }
 
-// Stop tells the Downloader to stop begin downloading bundles.
+func (d *Downloader) doStart(ctx context.Context) {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	d.wg.Add(1)
+	go d.loop(ctx)
+
+	for {
+		select {
+		case done := <-d.stop:
+			cancel()
+			d.wg.Wait()
+			done <- struct{}{}
+			return
+		}
+	}
```

Comment:
Hmm trying out alternative version of this, how about that?

```go
done := <-d.stop // blocks until there's something to read
cancel()
d.wg.Wait()
done <- struct{}{}
```

Do we need the infinite loop if the read on `d.stop` is blocking anyways? Once it returns something, we'll never try a second time either. 🤔 

Also since we're not interested in empty structs, we might as well `close(done)` here. It'll have the same effect, any (waiting) readers will get the type's zero value.

Reply:
Done